modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java

```diff
@@ -182,18 +192,32 @@ protected void doServeResource(
 		}
 	}
 
-	private JSONArray _getAssetCategoriesJSONArray(
-		ContentDashboardItem contentDashboardItem, Locale locale) {
+	private HashMap<String, Object> _buildVocabularyDetails(
+			Locale locale, AssetCategory assetCategory,
+			AssetVocabulary currentVocabulary)
+		throws PortalException {
 
-		List<AssetCategory> assetCategories =
-			contentDashboardItem.getAssetCategories();
+		return HashMapBuilder.<String, Object>put(
+			"categories",
+			() -> {
+				ArrayList<String> categoriesArray = new ArrayList<>();
 
-		Stream<AssetCategory> stream = assetCategories.stream();
+				categoriesArray.add(assetCategory.getTitle(locale));
 
-		return JSONUtil.putAll(
-			stream.map(
-				assetCategory -> assetCategory.getTitle(locale)
-			).toArray());
+				return categoriesArray;
+			}
+		).put(
+			"groupName",
+			ContentDashboardGroupUtil.getGroupName(
+				_groupLocalService.getGroup(currentVocabulary.getGroupId()),
```

Comment:
When we call the method getGroup, a portal exception can be thrown, it is not better to use fetchGroup in this situation?